contracts/TxRelay.sol

```diff
@@ -22,12 +25,17 @@ contract TxRelay {
         bytes32 sigR,
         bytes32 sigS,
         address destination,
-        bytes data
+        bytes data,
+        uint listNum
     ) public {
 
+        // only allow senders from the whitelist specified by the user,
+        // 0 means no whitelist.
+        require(listNum == 0 || whitelist[listNum][msg.sender]);
+
         address claimedSender = getAddress(data);
-        // relay :: nonce :: destination :: data :: relayer
-        bytes32 h = sha3(this, nonce[claimedSender], destination, data);
+        // relay :: nonce :: destination :: data :: whitelist
```

Comment:
Can we change this to:
`// relay :: whitelist :: nonce :: destination :: data`

Not functionality changing at all, but maybe cleaner (though this is so small, whatever :) )

Reply:
yeah, why not. We need to change eth-signer anyway.